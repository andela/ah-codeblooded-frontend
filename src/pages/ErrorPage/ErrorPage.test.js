import React from 'react';
import { shallow } from 'enzyme';
import { ErrorPage } from './index';

describe('<ErrorPage> component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<ErrorPage errorMessage="Page not found" errorCode={404} />);
    expect(wrapper).toMatchSnapshot();
  });
});
