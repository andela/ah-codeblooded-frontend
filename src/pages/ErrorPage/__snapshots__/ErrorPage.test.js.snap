// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorPage> component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorPage
    errorCode={404}
    errorMessage="Page not found"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="background"
        />,
        <div
          className="error-code center-align"
        >
          404
        </div>,
        <div
          className="row"
        >
          <div
            className="col m4 offset-m4"
          >
            <div
              className="card"
            >
              <div
                className="card-title center-align"
              >
                <h4>
                  Oops something went wrong!!
                </h4>
              </div>
              <div
                className="card-content center"
              >
                <div
                  className="error-message"
                >
                  Page not found
                </div>
              </div>
              <div
                className="card-action center"
              >
                <a
                  className="btn-bordered"
                  href="/"
                >
                  Take me home
                </a>
              </div>
            </div>
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "background",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 404,
          "className": "error-code center-align",
        },
        "ref": null,
        "rendered": 404,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col m4 offset-m4"
          >
            <div
              className="card"
            >
              <div
                className="card-title center-align"
              >
                <h4>
                  Oops something went wrong!!
                </h4>
              </div>
              <div
                className="card-content center"
              >
                <div
                  className="error-message"
                >
                  Page not found
                </div>
              </div>
              <div
                className="card-action center"
              >
                <a
                  className="btn-bordered"
                  href="/"
                >
                  Take me home
                </a>
              </div>
            </div>
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="card"
            >
              <div
                className="card-title center-align"
              >
                <h4>
                  Oops something went wrong!!
                </h4>
              </div>
              <div
                className="card-content center"
              >
                <div
                  className="error-message"
                >
                  Page not found
                </div>
              </div>
              <div
                className="card-action center"
              >
                <a
                  className="btn-bordered"
                  href="/"
                >
                  Take me home
                </a>
              </div>
            </div>,
            "className": "col m4 offset-m4",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="card-title center-align"
                >
                  <h4>
                    Oops something went wrong!!
                  </h4>
                </div>,
                <div
                  className="card-content center"
                >
                  <div
                    className="error-message"
                  >
                    Page not found
                  </div>
                </div>,
                <div
                  className="card-action center"
                >
                  <a
                    className="btn-bordered"
                    href="/"
                  >
                    Take me home
                  </a>
                </div>,
              ],
              "className": "card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h4>
                    Oops something went wrong!!
                  </h4>,
                  "className": "card-title center-align",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Oops something went wrong!!",
                  },
                  "ref": null,
                  "rendered": "Oops something went wrong!!",
                  "type": "h4",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="error-message"
                  >
                    Page not found
                  </div>,
                  "className": "card-content center",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Page not found",
                    "className": "error-message",
                  },
                  "ref": null,
                  "rendered": "Page not found",
                  "type": "div",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="btn-bordered"
                    href="/"
                  >
                    Take me home
                  </a>,
                  "className": "card-action center",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Take me home",
                    "className": "btn-bordered",
                    "href": "/",
                  },
                  "ref": null,
                  "rendered": "Take me home",
                  "type": "a",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="background"
          />,
          <div
            className="error-code center-align"
          >
            404
          </div>,
          <div
            className="row"
          >
            <div
              className="col m4 offset-m4"
            >
              <div
                className="card"
              >
                <div
                  className="card-title center-align"
                >
                  <h4>
                    Oops something went wrong!!
                  </h4>
                </div>
                <div
                  className="card-content center"
                >
                  <div
                    className="error-message"
                  >
                    Page not found
                  </div>
                </div>
                <div
                  className="card-action center"
                >
                  <a
                    className="btn-bordered"
                    href="/"
                  >
                    Take me home
                  </a>
                </div>
              </div>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "background",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 404,
            "className": "error-code center-align",
          },
          "ref": null,
          "rendered": 404,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col m4 offset-m4"
            >
              <div
                className="card"
              >
                <div
                  className="card-title center-align"
                >
                  <h4>
                    Oops something went wrong!!
                  </h4>
                </div>
                <div
                  className="card-content center"
                >
                  <div
                    className="error-message"
                  >
                    Page not found
                  </div>
                </div>
                <div
                  className="card-action center"
                >
                  <a
                    className="btn-bordered"
                    href="/"
                  >
                    Take me home
                  </a>
                </div>
              </div>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="card"
              >
                <div
                  className="card-title center-align"
                >
                  <h4>
                    Oops something went wrong!!
                  </h4>
                </div>
                <div
                  className="card-content center"
                >
                  <div
                    className="error-message"
                  >
                    Page not found
                  </div>
                </div>
                <div
                  className="card-action center"
                >
                  <a
                    className="btn-bordered"
                    href="/"
                  >
                    Take me home
                  </a>
                </div>
              </div>,
              "className": "col m4 offset-m4",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="card-title center-align"
                  >
                    <h4>
                      Oops something went wrong!!
                    </h4>
                  </div>,
                  <div
                    className="card-content center"
                  >
                    <div
                      className="error-message"
                    >
                      Page not found
                    </div>
                  </div>,
                  <div
                    className="card-action center"
                  >
                    <a
                      className="btn-bordered"
                      href="/"
                    >
                      Take me home
                    </a>
                  </div>,
                ],
                "className": "card",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h4>
                      Oops something went wrong!!
                    </h4>,
                    "className": "card-title center-align",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Oops something went wrong!!",
                    },
                    "ref": null,
                    "rendered": "Oops something went wrong!!",
                    "type": "h4",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="error-message"
                    >
                      Page not found
                    </div>,
                    "className": "card-content center",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Page not found",
                      "className": "error-message",
                    },
                    "ref": null,
                    "rendered": "Page not found",
                    "type": "div",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <a
                      className="btn-bordered"
                      href="/"
                    >
                      Take me home
                    </a>,
                    "className": "card-action center",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Take me home",
                      "className": "btn-bordered",
                      "href": "/",
                    },
                    "ref": null,
                    "rendered": "Take me home",
                    "type": "a",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
